### PR TITLE
Document hive.max-partitions-for-eager-load configuration property

### PR DESCRIPTION
**Pull request by colebow**
## Description

This adds docs for the config property added in #14225. Also a preliminary commit to reformat the table to a list-table, which is a functional no-op.

## Non-technical explanation

Docs only.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: docs/src/main/sphinx/connector/hive.rst [hunk 1]
```diff
@@ -339,9 +339,14 @@ Hive configuration properties
     * - ``hive.max-partitions-per-writers``
       - Maximum number of partitions per writer.
       - 100
+    * - ``hive.max-partitions-for-eager-load``
+      - The maximum number of partitions for a single table scan to haven loaded
+        eagerly on the coordinator. Certain optimizations are not possible
+        without eager loading.
+      - 100,000
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:346]: I think this should be 

```
100000
```

File: docs/src/main/sphinx/connector/hive.rst [hunk 2]
```diff
@@ -339,9 +339,14 @@ Hive configuration properties
...
     * - ``hive.max-partitions-per-scan``
       - Maximum number of partitions for a single table scan.
-      - 100,000
+      - 1,000,000
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:349]: 1000000

File: docs/src/main/sphinx/connector/hive.rst [hunk 3]
```diff
@@ -339,9 +339,14 @@ Hive configuration properties
     * - ``hive.max-partitions-per-writers``
       - Maximum number of partitions per writer.
       - 100
+    * - ``hive.max-partitions-for-eager-load``
+      - The maximum number of partitions for a single table scan to haven loaded
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:343]: to load eagerly on the coordinator.

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/hive.rst (hunk 2 above)
Comment (colebow) [reply docs/src/main/sphinx/connector/hive.rst:349]: This is consistent with all other properties on this table, and this is more readable considering the number of zeroes.

**Comment by martint** [comment]
Tested locally

**Pull request merged**